### PR TITLE
Gradle Cleanup

### PR DESCRIPTION
**Pull request by alonalbert**
* Added a version catalog
* Removed what seems to be unused dependencies
* Updated all libs to latest.

### CODE REVIEW

File: settings.gradle.kts [hunk 1]
```diff
@@ -5,16 +5,6 @@ pluginManagement {
...
     }
-
-    plugins {
-        kotlin("multiplatform").version(extra["kotlin.version"] as String)
-        id("org.jetbrains.compose").version(extra["compose.version"] as String)
-    }
-}
-
-plugins {
-    // Ensure JBR vendor is configured on CI, see https://github.com/actions/setup-java/issues/399.
```
Comment (romainguy) [inline settings.gradle.kts:16]: Don't remove, this is necessary to provide valid DMGs with Jewel on macOS

File: build.gradle.kts [hunk 2]
```diff
@@ -21,23 +21,18 @@ kotlin {
...
                 implementation(compose.desktop.currentOs) {
                     exclude(group = "org.jetbrains.compose.material")
                 }
-                implementation("org.jetbrains.jewel:jewel-int-ui-standalone-241:${extra["jewel.version"] as String}")
-                implementation("org.jetbrains.jewel:jewel-int-ui-decorated-window-241:${extra["jewel.version"] as String}")
-                implementation("org.jetbrains.skiko:skiko-awt-runtime-macos-arm64:${extra["skiko.version"] as String}")
-                implementation("org.jetbrains.compose.components:components-splitpane-desktop:${extra["compose.version"] as String}")
-                implementation("com.fifesoft:rsyntaxtextarea:${extra["rsyntaxtextarea.version"] as String}")
-                implementation("com.fifesoft:rstaui:${extra["rstaui.version"] as String}")
-                implementation("net.java.dev.jna:jna:${extra["jna.version"] as String}")
```
Comment (romainguy) [inline build.gradle.kts:37]: JNA is required by Skiko (at least on macOS). I had issues leaving the dependency out

**Review by romainguy — CHANGES_REQUESTED** [review]

File: settings.gradle.kts (hunk 1 above)
Comment (alonalbert) [reply settings.gradle.kts:16]: Done. Could not figure out how to integrate this with the version catalog.

File: build.gradle.kts (hunk 2 above)
Comment (alonalbert) [reply build.gradle.kts:37]: Done

**Review by romainguy — APPROVED** [review]
Thanks!

**Pull request merged**